test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Config
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class EdgeConfigBuilder
+    {
+        readonly string deviceId;
+        readonly Dictionary<string, IModuleConfigBuilder> moduleBuilders;
+        Option<(string address, string username, string password)> registry;
+
+        public EdgeConfigBuilder(string deviceId)
+        {
+            this.deviceId = deviceId;
+            this.moduleBuilders = new Dictionary<string, IModuleConfigBuilder>();
+            this.registry = Option.None<(string, string, string)>();
+        }
+
+        public void AddRegistryCredentials(string address, string username, string password)
+        {
+            this.registry = Option.Some((address, username, password));
+        }
+
+        public IModuleConfigBuilder AddEdgeAgent(string image = null)
+        {
+            Option<string> imageOption = Option.Maybe(image);
+            var builder = new AgentModuleConfigBuilder(imageOption);
+            this.moduleBuilders.Add(builder.Name, builder);
+            return builder;
+        }
+
+        public IModuleConfigBuilder AddEdgeHub(string image = null)
+        {
+            Option<string> imageOption = Option.Maybe(image);
+            var builder = new HubModuleConfigBuilder(imageOption);
+            this.moduleBuilders.Add(builder.Name, builder);
+            return builder;
+        }
+
+        public IModuleConfigBuilder AddModule(string name, string image)
+        {
+            var builder = new ModuleConfigBuilder(name, image);
+            this.moduleBuilders.Add(builder.Name, builder);
+            return builder;
+        }
+
+        public EdgeConfiguration Build()
+        {
+            // Build all modules *except* edge agent
+            List<ModuleConfiguration> modules = this.moduleBuilders
+                .Where(b => b.Key != "$edgeAgent")
+                .Select(b => b.Value.Build())
+                .ToList();
+
+            // Build edge agent
+            modules.Insert(0, this.BuildEdgeAgent(modules));
```

Comment:
Just want to confirm, if edge agent module is not the first module, will it cause problem?